WordPress/gutenberg
# Style Book: Move iframe to root of content area to support styles that overflow block previews

## What?


Fixes: https://github.com/WordPress/gutenberg/issues/48392

Alternative to #48470, this PR looks at moving the iframe for the Style Book up to the root of the content area, and uses editor context / block list at the example level, so that the styles affect and are displayed across the whole content area of the Style Book rather than just within each particular block example.

## Why?


Styles need to affect an area beyond just their preview — the idea in this PR is that (hopefully) we can get the styles to affect the entire content area so that it more closely resembles the layout of the editor.

## How?


* Add an iframe at the root of the content area of the style book
* Add some hard-coded styles to pass to that iframe along with editor styles
* At the individual example level, use a block editor context provider and block list for rendering the block previews
* Reset button styles and then include hard-coded rules for the button element, copying/pasting rules from the Button component as needed (e.g. focus styles)

Note / caveat: because of lifting the iframe up to the root of the content area, I had to set the iframe to `tabIndex=0` so that the buttons within it could be tabbed to. This seems to work pretty well, but means that the iframe itself now sits within the tab sequence of the style book. Happy for any suggestions on how to improve this.

## Testing Instructions





1. Open the Style Book and select Button under Design.
2. Adjust the Shadow settings for the Button block, and note that with this PR applied, the bigger shadows should still be visible outside of the area of the block example.
3. Double-check that other styles are appearing correctly, too, e.g. set the Image block to have a Duotone value set. The Duotone styling should be visible in the Style Book, too.
4. Try with a variety of themes with custom fonts, etc — check that the styling looks correct in the Style Book / matches `trunk`, and that the block headings still match the styling in Gutenberg

For code review: double-check whether or not I've included all the bits required from the block previews over to the examples here.

### Testing Instructions for Keyboard

1. Open the Style Book within global styles.
2. From this point tab to the examples within the Style Book content area. You should be able to tab between / press Enter to select any of the previews as on trunk. The main difference is that the iframe container area is now in the tab sequence. Again, happy for any ideas on how to improve this / skip the iframe in the tab sequence!

## Screenshots or screencast 

Button block with a large shadow applied in a site with a gradient set as the site's background color:

![image](https://user-images.githubusercontent.com/14988353/223301203-49b6db36-f0a0-4bd7-b160-c8d6a8d6c6e2.png)
